src/js/player.js

```diff
@@ -4290,6 +4300,69 @@ class Player extends Component {
     return !!this.isAudio_;
   }
 
+  /**
+   * Get the current audioOnlyMode state or set audioOnlyMode to true or false.
+   *
+   * Setting this to `true` will hide all player components except the control bar,
+   * as well as control bar components needed only for video.
+   *
+   * @param {boolean} [value]
+   *         The value to set audioOnlyMode to.
+   *
+   * @return {boolean}
+   *         True if audioOnlyMode is on, false otherwise.
+   */
+  audioOnlyMode(value) {
+    if (typeof value !== 'boolean' || value === this.audioOnlyMode_) {
+      return this.audioOnlyMode_;
+    }
+
+    if (value === true) {
+      // Update styling immediately so we can get the control bar's height
+      this.addClass('vjs-audio-only-mode');
+
+      const playerChildren = this.children();
+      const controlBar = this.getChild('ControlBar');
+      const controlBarHeight = controlBar && controlBar.height();
+
+      // Hide all player components except the control bar. Control bar components
+      // needed only for video are hidden with CSS
+      playerChildren.forEach(child => {
+        if (child === controlBar) {
+          return;
+        }
+
+        if (child.el_ && !child.hasClass('vjs-hidden')) {
+          child.hide();
+
+          this.audioOnlyCache_.hiddenChildren.push(child);
+        }
+      });
+
+      this.audioOnlyCache_.playerHeight = this.height();
+
+      // Set the player height the same as the control bar
+      this.height(controlBarHeight);
+
+      // Fullscreen is not supported in audioOnlyMode, so exit if we need to
+      if (this.isFullscreen()) {
+        this.exitFullscreen();
+      }
```

Comment:
should we also exit PiP?

Reply:
should we exit fs/pip before getting the control bar height, in case it's different in fullscreen? At least exitFullscreen should have a promise for knowing when exiting has finished.